sdk/python/feast/feature_logging.py

```diff
@@ -0,0 +1,167 @@
+    """
+
+    @abc.abstractmethod
+    def get_schema(self, registry: "Registry") -> pa.Schema:
+        """ Generate schema for logs destination. """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_partition_column(self, registry: "Registry") -> str:
+        """ Return partition column that must exist in generated schema. """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_log_timestamp_column(self) -> str:
+        """ Return timestamp column that must exist in generated schema. """
+        raise NotImplementedError
+
+
+class FeatureServiceLoggingSource(LoggingSource):
+    def __init__(self, feature_service: "FeatureService", project: str):
+        self._feature_service = feature_service
+        self._project = project
+
+    def get_schema(self, registry: "Registry") -> pa.Schema:
+        fields: Dict[str, pa.DataType] = {}
+
+        for projection in self._feature_service.feature_view_projections:
+            for feature in projection.features:
+                fields[
+                    f"{projection.name_to_use()}__{feature.name}"
+                ] = FEAST_TYPE_TO_ARROW_TYPE[feature.dtype]
+                fields[
+                    f"{projection.name_to_use()}__{feature.name}__timestamp"
+                ] = PA_TIMESTAMP_TYPE
+                fields[
+                    f"{projection.name_to_use()}__{feature.name}__status"
+                ] = pa.int32()
+
+            try:
+                feature_view = registry.get_feature_view(projection.name, self._project)
+            except FeatureViewNotFoundException:
+                try:
+                    on_demand_feature_view = registry.get_on_demand_feature_view(
+                        projection.name, self._project
+                    )
+                except OnDemandFeatureViewNotFoundException:
+                    raise FeastObjectNotFoundException(
+                        f"Can't recognize feature view with a name {projection.name}"
+                    )
+
+                for (
+                    request_source
+                ) in on_demand_feature_view.source_request_sources.values():
+                    for field in request_source.schema:
+                        fields[field.name] = FEAST_TYPE_TO_ARROW_TYPE[field.dtype]
+
+            else:
+                for entity_name in feature_view.entities:
+                    entity = registry.get_entity(entity_name, self._project)
+                    join_key = projection.join_key_map.get(
+                        entity.join_key, entity.join_key
+                    )
+                    fields[join_key] = FEAST_TYPE_TO_ARROW_TYPE[
+                        from_value_type(entity.value_type)
+                    ]
+
+        # system columns
+        fields["request_id"] = pa.string()
+        fields["log_timestamp"] = pa.timestamp("us", tz=UTC)
+        fields["log_date"] = pa.date32()
```

Comment:
Do we have a list of "system columns" blacklisted somewhere when we define features? How do you avoid column collisions?

Reply:
Good point. We use full feature names here, so collisions with feature names is unlikely. However, there might be conflicts with entities or request data. So I will add prefixes to this names.

Reply:
Added prefixes to system fields and put them into constants.

Reply:
I cannot find any list of forbidden names for entities (although we already have some system names, like `__dummy_entity`), so it's probably make sense to create an issue and address it in subsequent PR.